python-sdk/src/astro/sql/operators/cleanup.py

```diff
@@ -113,6 +121,27 @@ def drop_table(self, table: BaseTable) -> None:
         self.log.info("Dropping table %s", table.name)
         db.drop_table(table)
 
+    def _is_task_failed(self, task_instances: list[TaskInstance]) -> bool:
+        """
+        Given a list of task instances, return True if at least one task has failed.
+
+        :param task_instances:
+        :return: boolean if at least one task besides this one has failed
+        """
+        failed_tasks = [
+            (ti.task_id, ti.state)
+            for ti in task_instances
+            if ti.task_id != self.task_id and ti.state == State.FAILED
+        ]
+        if failed_tasks:
+            self.log.info(
```

Comment:
This log is  better placed in the `execute` method as part of the `self.skip_on_failure` - since that's where we'll have the skipping behaviour

Reply:
I may push back, as moving the log to `execute` means that `has_task_failed` would have to return the a list of failed tasks instead of a simple boolean;

Also, other cleanup functions `drop_table()` and `_is_dag_running()` have logging behavior inside them.